merlin/analysis/filterbarcodes.py

```diff
@@ -316,7 +338,7 @@ def extreme_values(inputData: pandas.Series):
                         codingCounts, 'coding_counts', self)
 
 
-class AdaptiveFilterBarcodes(decode.BarcodeSavingParallelAnalysisTask):
+class AdaptiveFilterBarcodes(FilterBarcodes):
```

Comment:
This creates extra 'area_threshold', 'intensity_threshold', and 'distance_threshold' parameters in the parameter set for AdaptiveFilterBarcode that aren't relevant. I would prefer creating another parent class called AbstractFilterBarcodes that defines the shared parameters and can be used as the base class for both FilterBarcodes and AdaptiveFilterBarcodes.  The  _remove_zplane_duplicates function can all be moved to the utility module since I don't think the cleanup_across_z function is especially useful without this function.